### PR TITLE
DSNS-229, 230 and 231

### PR DESCRIPTION
**Pull request by gordonyeong**
DSNS-231:
        - Introduced integration tests for landset l1 data - R3.3 Filter out
          the l1 scene if there is already an ARD scene with the same scene_id,
          that is un-archived
    DSNS-230:
        1. Introduced new integration tests for s2 data - R3.2 Process a scene
           if the child is interim and ancill data is there
        2. Updated util.py's get_extracted_path() to work with s2. Also tested
           the existing tests to ensure they still work.
    DSNS-229:
        - Introduced new integration tests for s2 data - R1.1 Unfiltered scenes
        are ARD processed

### CODE REVIEW

**Review by dunkgray — CHANGES_REQUESTED** [review]
here's some suggested modifications
https://github.com/GeoscienceAustralia/dea-ard-scene-select/pull/78
Though this branch has had a forced update, so merging could be an issue

**Comment by gordonyeong** [comment]
the only major difference i can see in the feature branch you have set up is just that the calls to scene select do not have `--config`, to which i didn't do because i did not want to break consistency. The original s2_go_select.sh and ls_go_select.sh did have an explicitly defined config to use.

**Comment by gordonyeong** [comment]
adjustements made as discussed :) kindly review when you are ready, Duncan :)

**Review by dunkgray — CHANGES_REQUESTED** [review]
lets remove the file - tests/integration_tests/test_s2_scenes_r1_2.py from this PR. Potentially the requirement R1.2: S2 scenes can be processed when the zip and yaml are in different directories doesn't fit with these tests.
why is tests/integration_tests/test_scene_r_2_7.py and tests/integration_tests/test_scene_r_2_6.py being deleted?

**Comment by gordonyeong** [comment]
lets remove the file - tests/integration_tests/test_s2_scenes_r1_2.py from this PR. Potentially the requirement R1.2: S2 scenes can be processed when the zip and yaml are in different directories doesn't fit with these tests.

  *   I can do this but does that mean we cancel DSNS-239 ?


________________________________

**Comment by gordonyeong** [comment]
> lets remove the file - tests/integration_tests/test_s2_scenes_r1_2.py from this PR. Potentially the requirement R1.2: S2 scenes can be processed when the zip and yaml are in different directories doesn't fit with these tests.



Does this mean that DSNS-239 is to be cancelled?

**Comment by dunkgray** [comment]
re: lets remove the file - tests/integration_tests/test_s2_scenes_r1_2.py from this PR. Potentially the requirement R1.2: S2 scenes can be processed when the zip and yaml are in different directories doesn't fit with these tests.

Does this mean that DSNS-239 is to be cancelled?

potentially, but let's get the .py out of this PR first

File: tests/integration_tests/test_s2_r3_2.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    )
+
+    assert (
+        result.exit_code == 0
+    ), f"The scene_select process failed to execute: {result.output}"
+    assert result.output != "", f" the result output is {result.output}"
+
+    # Use glob to search for the log file
+    # within filter-jobid-* directories
+    matching_files = list(Path(tmp_path).glob("filter-jobid-*/" + GEN_LOG_FILE))
```
Comment (dunkgray) [inline tests/integration_tests/test_s2_r3_2.py:152]: lines 150 - 152 can go.

**Review by dunkgray — CHANGES_REQUESTED** [review]
why is tests/integration_tests/test_scene_r_2_7.py being deleted?
Also, if commits are kept separate it help's me understand what the change is. So let's have individual commits. They can get squashed in the merge.

**Comment by gordonyeong** [comment]
that was deleted because it was a copy of
  tests/integration_tests/test_scene_r_2_6.py.I squashed them in readiness of a final review. Alright, i will leave the commits in their original forms.


________________________________

File: tests/integration_tests/test_s2_r3_2.py (hunk 1 above)
Comment (gordonyeong) [reply tests/integration_tests/test_s2_r3_2.py]: sorted.

**Comment by dunkgray** [comment]
> deleted because it was a copy of
  tests/integration_tests/test_scene_r_2_6.py
ok, thanks.

**Review by dunkgray — APPROVED** [review]

**Pull request merged**